### ISSUE TITLE
oauth problem with php7

### ISSUE THREAD

**Issue by thmo** [cweiske/grauphel#38]
(Moved comment from #36 into a new issue.)

Hm, here it doesn't work anymore since updating to php7.

Debian 8 with php 7.0.5-1 from dotdeb, owncloud 9.0.1, grauphel 0.6.2, oauth 2.0.1 installed via pecl, and tomboy 1.15.4 (on Fedora).

A call to /index.php/apps/grauphel/oauth/request_token gets a 400 Bad Request, with this body

```
94
oauth_problem=parameter_absent&oauth_parameters_absent=oauth_consumer_key%26oauth_signature%26oauth_signature_method%26oauth_nonce%26oauth_timestamp
0
```

although the request itself contained all these fields:

```
POST /index.php/apps/grauphel/oauth/request_token HTTP/1.1
Authorization: OAuth realm="Snowy",oauth_callback="http%3A%2F%2Flocalhost%3A8000%2Ftomboy-web-sync%2F",oauth_consumer_key="anyone",oauth_nonce="...",oauth_signature="...",oauth_signature_method="HMAC-SHA1",oauth_timestamp="...",oauth_version="1.0"
Content-Type: application/json
Content-Length: 0
Host: ...
```

Any idea what's going wrong here?

**Comment by cweiske** [comment]
- What web server do you use?
- How is PHP connected to your web server?
- To let me debug this further, create a file `phpinfo.php` in the ownCloud root directory with the following content:

`<?php phpinfo(); ?>`

Then send the URL to me; cweiske+grauphel@cweiske.de

**Comment by thmo** [comment]
> - What web server do you use?

nginx 1.8.1

> - How is PHP connected to your web server?

php-fpm

**Comment by cweiske** [comment]
Looking at the phpinfo file; all seems fine. Authorization header is available at the standard location of `HTTP_AUTHORIZATION`.
My next suspicion is a bug in pecl's oauth extension, but I have to debug that deeper.

**Comment by cweiske** [comment]
I can reproduce this locally. This is most likely a bug in pecl's oauth package. I'll make a minimal test script and open a bug report.

Bug report: https://bugs.php.net/bug.php?id=72006

**Comment by cweiske** [comment]
@thmo: There is a patch in git for pecl's oauth package. Could you try it? See https://bugs.php.net/bug.php?id=72006#1462854212

**Comment by thmo** [comment]
@cweiske: Can you point me to the patch? Can't find the commit 89c8886b... mentioned in [php bug 72006](https://bugs.php.net/bug.php?id=72006#1462854212) - the latest I see [in PHP's git](http://git.php.net/?p=pecl/web_services/oauth.git;a=summary) is [06c84c18](http://git.php.net/?p=pecl/web_services/oauth.git;a=commit;h=06c84c18d18b402e4401ef8f438cdd953b2f385f).

**Comment by cweiske** [comment]
I did not find it either. Left a comment at bugs.php.net

**Comment by Sean-Der** [comment]
Hey! Sorry about that I only pushed the patch to my personal repo https://github.com/Sean-Der/pecl-web_services-oauth/commit/89c8886b88972abb266db8f2cb126eeaa2718262

I will also push this to php.net, and when I get some verification that this patch fixes things I will make a new release.

thanks!

**Comment by thmo** [comment]
@cweiske @Sean-Der I can confirm that the patch fixes the issue for me. Thanks!

**Comment by cweiske** [comment]
Very good. I'm closing this now.

@Sean-Der: Could you please release a new oauth version?

**Comment by Sean-Der** [comment]
Oof sorry, I really fell behind on this!

I released 2.0.2 with the patch, if anything is broken feel free to mention me here.
